src/test/java/teammates/sqlui/webapi/GetStudentActionTest.java

```diff
@@ -0,0 +1,288 @@
+import teammates.storage.sqlentity.Team;
+import teammates.ui.output.StudentData;
+import teammates.ui.webapi.GetStudentAction;
+
+/**
+ * SUT: {@link GetStudentAction}.
+ */
+public class GetStudentActionTest extends BaseActionTest<GetStudentAction> {
+    private Student stubStudent;
+    private Course stubCourse;
+    private Instructor stubInstructor;
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.STUDENT;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return GET;
+    }
+
+    @BeforeMethod
+    protected void setUp() {
+        stubStudent = getTypicalStudent();
+        Team stubTeam = getTypicalTeam();
+        stubStudent.setTeam(stubTeam);
+        stubStudent.setRegKey("RANDOM_KEY");
+        stubCourse = getTypicalCourse();
+        stubInstructor = getTypicalInstructor();
+    }
+
+    @Test
+    void testExecute_emptyParams_throwsInvalidHttpParameterException() {
+        String[] params = {};
+        verifyHttpParameterFailure(params);
+    }
+
+    @Test
+    void testExecute_validParamsUnregisteredStudentNotLoggedIn_success() {
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, stubCourse.getId(),
+                Const.ParamsNames.REGKEY, stubStudent.getRegKey(),
+        };
+        when(mockLogic.getStudentByRegistrationKey(stubStudent.getRegKey())).thenReturn(stubStudent);
+        GetStudentAction action = getAction(params);
+        StudentData studentData = (StudentData) getJsonResult(action).getOutput();
+        assertEquals(studentData.getEmail(), stubStudent.getEmail());
+        assertEquals(studentData.getKey(), null);
+    }
+
+    @Test
+    void testExecute_invalidRegKeyParamsUnregisteredStudent_throwsEntityNotFoundException() {
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, stubCourse.getId(),
+                Const.ParamsNames.REGKEY, "INVALID_KEY",
+        };
+        verifyEntityNotFound(params);
+
+        String[] params2 = {
+                Const.ParamsNames.COURSE_ID, stubCourse.getId(),
+        };
+        verifyEntityNotFound(params2);
+
+        String[] params3 = {
+                Const.ParamsNames.COURSE_ID, stubCourse.getId(),
+                Const.ParamsNames.REGKEY, null,
+        };
+
+        verifyEntityNotFound(params3);
+    }
+
+    @Test
+    void testExecute_validParamsRegisteredStudentLoggedIn_success() {
+        loginAsStudent(stubStudent.getGoogleId());
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, stubCourse.getId(),
+        };
+        when(mockLogic.getStudentByGoogleId(stubCourse.getId(), stubStudent.getGoogleId())).thenReturn(stubStudent);
+        GetStudentAction action = getAction(params);
+        StudentData studentData = (StudentData) getJsonResult(action).getOutput();
+        assertEquals(studentData.getEmail(), stubStudent.getEmail());
+        assertEquals(studentData.getKey(), null);
```

Comment:
maybe we can follow the old test case and create a private method for checking if the student data matches?